### PR TITLE
Magn 8758 view extensions localized loading

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

PR for:

MAGN-8758 Publish Customizer application menu items absent in French Windows
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8758

This PR addresses the issue that view extensions were not loaded in localized environments. This was due to a search for the file menu using the string ```"_file"```. Instead we extract the localized header of the file menu from the properties.resources dictionary and compare against this.

This is done for other menus that view extensions might add items to by creating an extension method that extracts the appropriate localized string depending on menu type.

loading now works on EN-US and FR

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@ramramps 

### FYIs

@jnealb

### CODE REVIEW

File: src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs [hunk 1]
```diff
@@ -118,4 +121,31 @@ public enum MenuBarType
         View,
         Help
     }
+
+    /// <summary>
+    /// A class of extensions for the MenuBarType Enum.
+    /// </summary>
+    public static class MenuBarTypeExtensions
```
Comment (pboyer) [inline src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs]: Can this be internal?

File: src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs [hunk 2]
```diff
@@ -118,4 +121,31 @@ public enum MenuBarType
...
+    {
+        /// <summary>
+        /// A method to extract the appropriate localized string 
+        /// representing the header name of this menu type
+        /// i.e. file -> "_File"
+        /// </summary>
+        /// <param name="type"></param>
+        /// <returns>A localized string used as the menu header</returns>
+        public static string ToDisplayString(this MenuBarType type)
```
Comment (ramramps) [inline src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs]: Can the extensions be on a different file?

**Comment by ramramps** [comment]
one small change, then LGTM.

**Comment by mjkkirschner** [comment]
@ramramps I've made the class internal and moved to a new file - please take on more look.

**Pull request merged**